vterm-module-make.el

```diff
@@ -2,22 +2,44 @@
 
 (require 'files)
 
-(defvar vterm-install-buffer-name " *Install vterm"
+(defvar vterm-install-buffer-name "*Install vterm*"
   "Name of the buffer used for compiling vterm-module.")
 
+(defcustom vterm-compile-with-system-libvterm nil
+  "Compile vterm with the system libvterm instead of downloading
+and a local copy from GitHub."
+  :type  'boolean
+  :group 'vterm)
+
 ;;;###autoload
 (defun vterm-module-compile ()
-  "This function compiles the vterm-module."
+  "This function compiles the vterm-module.
+
+If the variable vterm-compile-with-system-libvterm is set to t,
+the system libvterm is used."
   (interactive)
-  (let ((default-directory
-          (file-name-directory (file-truename (locate-library "vterm"))))
-        (make-commands
-         "mkdir -p build; \
-          cd build; \
-          cmake \
-            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-            ..; \
-          make"))
+  (let* ((vterm-directory
+           (file-name-directory (file-truename (locate-library "vterm"))))
+         ;; Ask the user if they want to compile with the version of libvterm
+         ;; on the system. It has to be available! (We don't perform checks)
+         (compile-with-system-libvterm
+          (if (or
+               vterm-compile-with-system-libvterm
```

Comment:
I feel like some people will inevitably get prompted every time they install/update the package if we use a binary option like this, and that would quickly get annoying. I'd suggest that the var should be ternary, with an `'ask` option which triggers prompting. The default would either be `ask`, or an option that works in most cases (as the default currently does nicely for me).